richeditor-compose/src/commonMain/kotlin/com/mohamedrejeb/richeditor/model/RichTextState.kt

```diff
@@ -1735,23 +1753,25 @@ public class RichTextState internal constructor(
             // If the new paragraph is empty apply style depending on the config
             if (tempTextFieldValue.selection.collapsed && newParagraph.isEmpty()) {
                 val newParagraphFirstRichSpan = newParagraph.getFirstNonEmptyChild()
-
-                val isSelectionAtNewRichSpan =
-                    newParagraphFirstRichSpan?.textRange?.min == tempTextFieldValue.selection.min - 1
-
-                // Check if the cursor is at the new paragraph
-                if (
-                    (!config.preserveStyleOnEmptyLine || richSpan.paragraph.isEmpty()) &&
-                    isSelectionAtNewRichSpan
-                ) {
-                    newParagraphFirstRichSpan.spanStyle = SpanStyle()
-                    newParagraphFirstRichSpan.richSpanStyle = RichSpanStyle.Default
-                } else if (
-                    config.preserveStyleOnEmptyLine &&
-                    isSelectionAtNewRichSpan
-                ) {
-                    newParagraphFirstRichSpan.spanStyle = currentSpanStyle
-                    newParagraphFirstRichSpan.richSpanStyle = currentRichSpanStyle
+                if (newParagraphFirstRichSpan != null) {
```

Comment:
This nullable var wasn't being handled as such, so i had to add in a null check. But CI and things pass without this, i dont know what is going on.